cli: Set user_agent header

wfchandler opened the pull request:
Our newly introduced audit log captures the user agent of the client, but currently the CLI does not set this header.

Add an option to configure the user agent to the SDK and set it for the CLI.

Closes https://github.com/oxidecomputer/oxide.rs/issues/1179

wfchandler committed:
cli: Set user_agent header

Our newly introduced audit log captures the user agent of the client,
but currently the CLI does not set this header.

Add an option to configure the user agent to the SDK and set it for the
CLI.

Closes https://github.com/oxidecomputer/oxide.rs/issues/1179

ahl commented on sdk/src/auth.rs:
```diff
@@ -139,6 +141,12 @@ impl ClientConfig {
     }
 
+    /// Specify the user_agent header to be sent by the client.
+    pub fn with_user_agent(mut self, user_agent: String) -> Self {
```
might do what we do above in `with_resolve`
```suggestion
    pub fn with_user_agent(mut self, user_agent: impl ToString) -> Self {
```

wfchandler replied:
Good call, added.

ahl reviewed (approved):
do we want the SDK to set a default agent that includes the API version e.g. "Rust SDK {}" and/or we could (probably?) capture the name of the binary.

wfchandler committed:
Set default user agent and take ToString

david-crespo reviewed (approved):
I had a similar question to Adam — I agree that giving the SDK its own user agent probably makes sense.

<img width="799" height="319" alt="Image" src="https://github.com/user-attachments/assets/c4645e51-f083-42b8-bd43-db21deb11ff1" />

wfchandler committed:
Update format and remove Option

Update the default and CLI user agents to use the typical
`<package>/<version> (<notes>)` format generally used. Also, remove the
`Option` on the user_agent field, as it's always set now.

wfchandler committed:
Fix client_builder to directly take user_agent

wfchandler commented:
> do we want the SDK to set a default agent that includes the API version e.g. "Rust SDK {}" and/or we could (probably?) capture the name of the binary.

Good idea, I've set the default to the package name and version.
